src/widget/wtracktableview.cpp

```diff
@@ -663,6 +667,40 @@ void WTrackTableView::slotOpenInFileBrowser() {
     }
 }
 
+void WTrackTableView::slotRemoveFromDisk() {
+       TrackModel* trackModel = getTrackModel();
+    if (!trackModel) {
+        return;
+    }
+    QMessageBox msgBox (QMessageBox::Information, QObject::tr("Remove From Disk"), QObject::tr("Really delete the files from disk? (Permanent Deletion)"));
+    QAbstractButton* pYES = (QAbstractButton*)msgBox.addButton(QMessageBox::Yes);
+    QAbstractButton* pNO  = (QAbstractButton*)msgBox.addButton(QMessageBox::No);
+    QCheckBox alsoHide(QObject::tr("Also hide track in library"), &msgBox);
+    alsoHide.blockSignals(true);
+    msgBox.addButton(&alsoHide, QMessageBox::ActionRole);
+
+    if (QMessageBox::Yes == msgBox.exec())
+    {
+
+
+        QModelIndexList indices = selectionModel()->selectedRows();
+
+        for (const QModelIndex& index : indices) {
+            if (!index.isValid()) {
+              continue;
+            }
+            QString  filenameWithPath = trackModel->getTrackLocation(index);
+            QFile file (filenameWithPath);
+            file.remove();
+
+        }
+        if( alsoHide.checkState() == Qt::Checked )
+        {
+            trackModel->hideTracks(indices);
```

Comment:
shouldn't this be a purge? if the file is gone we should probably permanently remove the metadata too

Reply:
That depends on the desired use case.
I will try to propose texts to distinguish it.

Reply:
Tried to purge using trackModel->purgeTracks but.. it didn't work. not sure why. Even did hideTracks then purgeTracks. They stayed in hidden. So.. kinda stumped.